src/txmempool.cpp

```diff
@@ -99,29 +135,70 @@ bool CTxMemPool::addUnchecked(const uint256& hash, const CTxMemPoolEntry &entry,
     LOCK(cs);
     mapTx[hash] = entry;
     const CTransaction& tx = mapTx[hash].GetTx();
-        for (unsigned int i = 0; i < tx.vin.size(); i++)
-            mapNextTx[tx.vin[i].prevout] = CInPoint(&tx, i);
+
+    for (unsigned int i = 0; i < tx.vin.size(); i++)
+        mapNextTx[tx.vin[i].prevout] = CInPoint(&tx, i);
+
     BOOST_FOREACH(const JSDescription &joinsplit, tx.vjoinsplit) {
         BOOST_FOREACH(const uint256 &nf, joinsplit.nullifiers) {
             mapNullifiers[nf] = &tx;
         }
     }
+
+    for(const auto& sc: tx.vsc_ccout) {
+        assert(mapSidechains[sc.scId].scCreationTxHash.IsNull());
+        mapSidechains[sc.scId].scCreationTxHash = hash;
+    }
+
+    for(const auto& fwd: tx.vft_ccout) {
+        mapSidechains[fwd.scId].CcTransfersSet.insert(hash);
+    }
+
     nTransactionsUpdated++;
     totalTxSize += entry.GetTxSize();
     cachedInnerUsage += entry.DynamicMemoryUsage();
     minerPolicyEstimator->processTransaction(entry, fCurrentEstimate);
+    LogPrint("sc", "%s():%d - tx [%s] added in mempool\n", __func__, __LINE__, hash.ToString() );
+
+    return true;
+}
 
+bool CTxMemPool::addUnchecked(const uint256& hash, const CCertificateMemPoolEntry &entry, bool fCurrentEstimate)
+{
+    LOCK(cs);
+    mapCertificate[hash] = entry;
+    // no mapNextTx or mapNullifiers handling is necessary for certs since they have no inputs or joinsplits 
+
+    const CScCertificate& cert = mapCertificate[hash].GetCertificate();
+
+    mapSidechains[cert.scId].CcTransfersSet.insert(hash);
```

Comment:
We discussed about this choice AFK. I'll note down one issue with the idea of keeping both certificates and txs hashes in the same set, which emerged during the merge.
At some point we may want to return the ScInfo for a sidechain including the balance potential changes due to fwds and certs in the mempool. It would be easier to do so if certificates were not in the same set as txs. Consider that, afaik, there could be at most one certificate in mempool for a given sc. Once the merge is done, I'd really like to store this one certificate in a different variable that CcTransfersSet